src/Init/Control/Basic.lean

```diff
@@ -20,8 +20,21 @@ def Functor.discard {f : Type u → Type v} {α : Type u} [Functor f] (x : f α)
 
 export Functor (discard)
 
+/--
+An `Alternative` functor is an `Applicative` functor that has a notion of “emptyness” or “failure”
+and a binary operation `<|>` with a notion of “collecting values” or “left-most success”.
+
+Important instances include
+* `Option`, where `failure := .none` and `<|>` returns the left-most `.some`.
+* Parser combinators typically provide an `Applicative` instance for error-handling and
+  backtracking.
+-/
+-- NB: List instance is in mathlib. Once upstreamed, add
+-- * `List`, where `failure` is the empty list and `<|>` concatenates.
 class Alternative (f : Type u → Type v) extends Applicative f : Type (max (u+1) v) where
+  /-- See `Alternative`. -/
   failure : {α : Type u} → f α
+  /-- See `Alternative`. Can be writen using `<|>`.  -/
```

Comment:
```suggestion
  /--
  Depending on the `Alternative` instance, collects values or recovers from `failure`s by
  returning the leftmost success.
  -/
```

Reply:
Did you intentionally drop the reference to the binary operator syntax?

Reply:
The dangers of commenting before coffee...  The syntax should definitely be there.

Reply:
```suggestion
  /--
  Depending on the `Alternative` instance, collects values or recovers from `failure`s by
  returning the leftmost success. Can be written using the `<|>` operator syntax.
  -/
```